store/src/sql/entities/event.rs

```diff
@@ -129,13 +151,71 @@ impl EventInsertableBody {
         }
         let roots: BTreeSet<Cid> = reader.header().roots().iter().cloned().collect();
         let mut idx = 0;
-        let mut blocks = vec![];
+        let mut blocks = vec![]; // use a hashset and store the CID instead of recalculating?
         while let Some((cid, data)) = reader.next_block().await.map_err(Error::new_app)? {
             let ebr = EventBlockRaw::try_new(&event_cid, idx, roots.contains(&cid), cid, data)
                 .map_err(Error::from)?;
             blocks.push(ebr);
             idx += 1;
         }
-        Ok(Self::new(event_cid, blocks))
+
+        let ev_block = blocks
```

Comment:
this is a really long function, can we pull this part out into a helper?

Reply:
Actually, can you just use `event_from_car` for all this parsing?  https://github.com/ceramicnetwork/rust-ceramic/blob/107be36d27c73ae96a07614f69503596c48d4f10/api/src/server/event.rs#L23

Reply:
yeah, this behavior should be in the event crate and not here. we're duplicating it in api and store to avoid circular dependencies as they originally had slightly different concerns but as we've hardened the Event types, it probably makes sense to provide this behavior in one place.